### PR TITLE
Adding predict call

### PR DESCRIPTION
**Pull request by maxhutch**
Still debugging.

CC: @sparadiso @jsomara

### CODE REVIEW

**Comment by maxhutch** [comment]
Ready for a look @kjaym 

Huge thanks to @sparadiso and @jsomara

File: citrination_client/client.py [hunk 1]
```diff
@@ -151,6 +151,31 @@ def _get_list(values):
...
+
+        url = self._get_predict_url(model_name)
+        body = pif.dumps(
+            {"predictionRequest": {"predictionSource": "scalar", "usePrior": True, "candidates": candidates}}
+        )
+
+        response = requests.post(url, data=body, headers=self.headers)
+        if response.status_code == requests.codes.ok:
+            return response.json()['candidates']
+        return None
```
Comment (kjaym) [inline citrination_client/client.py]: Raise an exception here?

File: citrination_client/client.py [hunk 2]
```diff
@@ -151,6 +151,31 @@ def _get_list(values):
...
+
+    def _get_predict_url(self, model_name):
+        return self.api_url + '/csv_to_models/' + model_name + '/predict'
```
Comment (kjaym) [inline citrination_client/client.py]: The predict api is only exposed for csv to models?

File: citrination_client/client.py [hunk 3]
```diff
@@ -151,6 +151,31 @@ def _get_list(values):
...
+            candidates = [candidates]
+
+        url = self._get_predict_url(model_name)
+        body = pif.dumps(
+            {"predictionRequest": {"predictionSource": "scalar", "usePrior": True, "candidates": candidates}}
+        )
+
+        response = requests.post(url, data=body, headers=self.headers)
+        if response.status_code == requests.codes.ok:
+            return response.json()['candidates']
```
Comment (kjaym) [inline citrination_client/client.py]: Why not return the entire response?

Comment (sparadiso) [inline citrination_client/client.py]: Do our users care about an HTTP response or the prediction results? I don't really know what your plans were for this, but I would expect the data, not a Response object.

File: citrination_client/client.py (hunk 2 above)
Comment (sparadiso) [inline citrination_client/client.py]: Yep - it will be exposed for views once those are live.

File: citrination_client/client.py (hunk 1 above)
Comment (sparadiso) [inline citrination_client/client.py]: Good point. What exception would you raise here? RuntimeError custom?

Comment (kjaym) [inline citrination_client/client.py]: Runtime is fine

File: citrination_client/client.py (hunk 2 above)
Comment (kjaym) [inline citrination_client/client.py]: Will there be one route? Or is there going to a different route for each method used to generate a model? That is, are we going to have /models/model_name/predict to cover all models, or are we going to have /csv_to_models/model_name/predict and /views/model_name/predict?

File: citrination_client/client.py (hunk 3 above)
Comment (kjaym) [inline citrination_client/client.py]: I meant `return response.json()`, not `return response`. Sorry, that was not clear.

**Comment by sparadiso** [comment]
Ah, I see. I was just unpacking the only key for them, but if we update
that api to return more information then this will hide it until we update
the client. I see the wisdom in returning the full response in that case.

File: citrination_client/client.py (hunk 1 above)
Comment (maxhutch) [inline citrination_client/client.py]: Done

File: citrination_client/client.py (hunk 3 above)
Comment (maxhutch) [inline citrination_client/client.py]: Done

**Pull request merged**